sycl/plugins/unified_runtime/ur/adapters/opencl/queue.cpp

```diff
@@ -0,0 +1,161 @@
+//===--------- memory.cpp - OpenCL Adapter ---------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===-----------------------------------------------------------------===//
+
+#include "common.hpp"
+#include "platform.hpp"
+
+cl_command_queue_info mapURQueueInfoToCL(const ur_queue_info_t PropName) {
+
+  switch (PropName) {
+  case UR_QUEUE_INFO_CONTEXT:
+    return CL_QUEUE_CONTEXT;
+  case UR_QUEUE_INFO_DEVICE:
+    return CL_QUEUE_DEVICE;
+  case UR_QUEUE_INFO_DEVICE_DEFAULT:
+    return CL_QUEUE_DEVICE_DEFAULT;
+  case UR_QUEUE_INFO_FLAGS:
+    return CL_QUEUE_PROPERTIES_ARRAY;
+  case UR_QUEUE_INFO_REFERENCE_COUNT:
+    return CL_QUEUE_REFERENCE_COUNT;
+  case UR_QUEUE_INFO_SIZE:
+    return CL_QUEUE_SIZE;
+  default:
+    return -1;
+  }
+}
+
+cl_command_queue_properties
+convertURQueuePropertiesToCL(const ur_queue_properties_t *URQueueProperties) {
+  cl_command_queue_properties CLCommandQueueProperties = 0;
+
+  if (URQueueProperties->flags & UR_QUEUE_FLAG_OUT_OF_ORDER_EXEC_MODE_ENABLE) {
+    CLCommandQueueProperties |= CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE;
+  }
+  if (URQueueProperties->flags & UR_QUEUE_FLAG_PROFILING_ENABLE) {
+    CLCommandQueueProperties |= CL_QUEUE_PROFILING_ENABLE;
+  }
+  if (URQueueProperties->flags & UR_QUEUE_FLAG_ON_DEVICE) {
+    CLCommandQueueProperties |= CL_QUEUE_ON_DEVICE;
+  }
+  if (URQueueProperties->flags & UR_QUEUE_FLAG_ON_DEVICE_DEFAULT) {
+    CLCommandQueueProperties |= CL_QUEUE_ON_DEVICE_DEFAULT;
+  }
+
+  return CLCommandQueueProperties;
+}
+
+UR_APIEXPORT ur_result_t UR_APICALL urQueueCreate(
+    ur_context_handle_t hContext, ur_device_handle_t hDevice,
+    const ur_queue_properties_t *pProperties, ur_queue_handle_t *phQueue) {
+
+  cl_platform_id CurPlatform;
+  CL_RETURN_ON_FAILURE_AND_SET_NULL(
+      clGetDeviceInfo(cl_adapter::cast<cl_device_id>(hDevice),
+                      CL_DEVICE_PLATFORM, sizeof(cl_platform_id), &CurPlatform,
+                      nullptr),
+      phQueue);
+
+  cl_command_queue_properties CLProperties =
+      convertURQueuePropertiesToCL(pProperties);
```

Comment:
```suggestion
-      convertURQueuePropertiesToCL(pProperties);
+      pProperties ? convertURQueuePropertiesToCL(pProperties) : 0;
```
pProperties is optional